packages/dd-trace/src/serverless.js

```diff
@@ -60,9 +60,18 @@ function getIsAzureFunction () {
   return isAzureFunction
 }
 
+function isInServerlessEnvironment () {
+  const inAWSLambda = process.env.AWS_LAMBDA_FUNCTION_NAME !== undefined
+  const isGCPFunction = getIsGCPFunction()
+  const isAzureFunction = getIsAzureFunction()
+
+  return inAWSLambda || isGCPFunction || isAzureFunction
```

Comment:
How about we have this folded into the final expression, so it short-circuits. (`getIsGCPFunction` could be rewritten so it short circuits too, possibly first evaluating `isNewerGCPFunction` as I imagine that's more frequent.) I figure microoptimizations are on topic in a PR that deals with startup time :-)

Reply:
Worth looking into, but for the purpose of this PR, existing code was just copied over so I would consider it out of scope.